packages/pwa-kit-create-app/scripts/create-mobify-app.js

```diff
@@ -710,24 +769,79 @@ const runGenerator = (context, {outputDir, templateVersion, verbose}) => {
-                .forEach((relFilePath) =>
+                .map((file) => {
+                    const relFilePath = file.replace(assetsDir, '')
+                    return relFilePath
+                })
+                .forEach((relFilePath) => {
                     processTemplate(relFilePath, assetsDir, outputDir, context)
+                })
+        }
+
+        let extensionSamplePath
+        if (extractExtension) {
+            const extensionTmp = fs.mkdtempSync(p.resolve(os.tmpdir(), 'extract-extension-sample'))
+            const extensionTarFile = sh
+                .exec(
+                    `npm pack @salesforce/extension-sample@1.0.0-dev --pack-destination="${extensionTmp}"`,
+                    {
+                        silent: true
+                    }
                 )
+                .stdout.trim()
+            const extensionTarPath = p.join(extensionTmp, extensionTarFile)
+
+            // Extract the extension
+            const extensionFolder = p.join(outputDir, 'extension-sample')
+            tar.x({
+                file: extensionTarPath,
+                cwd: extensionTmp,
+                sync: true
+            })
+
+            // Copy the extension-sample into the output directory
+            extensionSamplePath = p.join(extensionTmp, 'package')
+            sh.cp('-rf', extensionSamplePath, outputDir)
+            sh.mv(extensionSamplePath, extensionDir)
+
+            // Clean up extension temporary directory
+            sh.rm('-rf', extensionTmp)
+
+            // Set the relative path for the extracted extension
+            extensionSamplePath = `file:./extension-sample`
         }
 
-        // Update the generated projects version. NOTE: For bootstrapped projects this
+        // Update the generated project's package.json. NOTE: For bootstrapped projects this
         // can be done in the template building. But since we have two types of project builds,
         // (bootstrap/bundle) we'll do it here where it works in both scenarios.
         const pkgJsonPath = p.resolve(outputDir, 'package.json')
         const pkgJSON = readJson(pkgJsonPath)
+
+        // Add @salesforce/extension-sample to dependencies and mobify object with extensions
         const finalPkgData = merge(pkgJSON, {
             name: slugifyName(context.answers.project.name || context.preset.id),
-            version: GENERATED_PROJECT_VERSION
+            version: GENERATED_PROJECT_VERSION,
+            devDependencies: {
+                '@salesforce/extension-sample': extractExtension ? extensionSamplePath : '1.0.0-dev'
+            },
+            mobify: {
+                app: {
+                    extensions: [
+                        [
+                            '@salesforce/extension-sample',
+                            {
+                                path: '/foo-page'
+                            }
+                        ]
+                    ]
+                }
+            }
```

Comment:
This is good for now. But I think we might end up using config files for the typescript minimal project and that might prompt us to use the `/assets/templates` mechanism for the generator that allows us to use handlebars templates to replace code. 

We might want to talk about this for a bit, and maybe think about creating a ticket for it as a follow up.

Reply:
The extract vs no extract logic is getting complicated.
I don't think we can implement all the javascript logic into an equivalent handlebars template notation in a single `package.json.hbs` asset file.